karyon3-admin-simple/src/main/java/com/netflix/karyon/admin/rest/AdminHttpHandler.java

```diff
@@ -40,56 +46,63 @@ public AdminHttpHandler(
             @AdminServer ObjectMapper mapper,
             @AdminServer Provider<ResourceContainer> controllers, 
             AdminServerConfig config,
+            AdminUIServerConfig uiConfig,
             Config cfg) {
         this.resources = controllers;
         this.mapper = mapper;
         this.config = config;
         this.cfg = cfg;
+        this.provider = 
+                new CachingStaticResourceProvider(
+                    new FileSystemResourceProvider(
+                        uiConfig.resourcePath(),
+                        uiConfig.mimeTypesResourceName()));
     }
     
     @Override
     public void handle(HttpExchange exchange) throws IOException {
-        LOG.info("'{}'", exchange.getRequestURI());
+        LOG.debug("'{}'", exchange.getRequestURI());
 
         final String path = exchange.getRequestURI().getPath();
         
-        exchange.getResponseHeaders().set("Server", "KaryonAdmin");
-        
         try {
-            // Redirect the server root to the configured remote server using the naming convension
-            //  
-            if (path.equals("/")) {
-                String addr = new Interpolator(cfg).interpolate(config.remoteServer());
-                LOG.info("Redirecting to '{}'", addr);
-                exchange.getResponseHeaders().set("Location", addr);
-                exchange.sendResponseHeaders(302, 0);
-                exchange.close();
-            }
-            else if (path.equals("/favicon.ico")) {
-                exchange.sendResponseHeaders(404, 0);
-                exchange.close();
+            try {
```

Comment:
It seems that the nested try here could instead become just two catches on a single try with the more specific NotFoundException coming first.